[TEST-FAILURE] LockBasicDistributedTest.testTryLock

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OpenJDK6/549/com.hazelcast$hazelcast/testReport/com.hazelcast.concurrent.lock/LockBasicDistributedTest/testTryLock/

```

02:34:57,614 FATAL [Node] testTryLock - [127.0.0.1]:7284 [dev] [3.5-EA2-SNAPSHOT] Error while joining the cluster!
java.lang.AssertionError: Node[Address[127.0.0.1]:7284] should have been joined to Address[127.0.0.1]:7283
	at com.hazelcast.test.TestNodeRegistry$MockJoiner.doJoin(TestNodeRegistry.java:194)
	at com.hazelcast.cluster.impl.AbstractJoiner.join(AbstractJoiner.java:93)
	at com.hazelcast.instance.Node.join(Node.java:534)
	at com.hazelcast.instance.Node.start(Node.java:343)
	at com.hazelcast.instance.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:126)
	at com.hazelcast.instance.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:152)
	at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:135)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:80)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newInstances(TestHazelcastInstanceFactory.java:96)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newInstances(TestHazelcastInstanceFactory.java:102)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newInstances(TestHazelcastInstanceFactory.java:90)
	at com.hazelcast.concurrent.lock.LockBasicDistributedTest.newInstances(LockBasicDistributedTest.java:14)
	at com.hazelcast.concurrent.lock.LockBasicTest.setup(LockBasicTest.java:51)
	at sun.reflect.GeneratedMethodAccessor94.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at com.hazelcast.test.HazelcastParallelClassRunner.access$201(HazelcastParallelClassRunner.java:29)
	at com.hazelcast.test.HazelcastParallelClassRunner$TestRunner.run(HazelcastParallelClassRunner.java:89)
	at java.lang.Thread.run(Thread.java:662)
```

I believe issue is related with mock joiner since there is hardcoded 1000 times join retry in the mock joiner with 50 ms interval which sums up 50 seconds and the join in this particular test is slow. We can increase this retry count and see if this is going to fail again.

--- a/hazelcast/src/test/java/com/hazelcast/test/TestNodeRegistry.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/TestNodeRegistry.java
@@ -182,7 +182,7 @@ final class TestNodeRegistry {
                     node.setJoined();
                     node.setAsMaster();
                 } else {
-                    for (int i = 0; !node.joined() && i < 1000; i++) {
+                    for (int i = 0; !node.joined() && i < 2000; i++) {
                         try {
                             node.clusterService.sendJoinRequest(node.getMasterAddress(), true);
                             Thread.sleep(50);

